### PR TITLE
Add double script fields

### PR DESCRIPTION
**Pull request by nik9000**
This adds support for `double` typed `script` fields.

### CODE REVIEW

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldType.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        if (values.isEmpty()) {
+            return Queries.newMatchAllQuery();
+        }
+        LongSet terms = new LongHashSet(values.size());
+        for (Object value : values) {
+            terms.add(Double.doubleToLongBits(NumberType.objectToDouble(value)));
+        }
+        checkAllowExpensiveQueries(context);
+        return new DoubleScriptFieldTermsQuery(script, leafFactory(context), name(), terms);
+    }
```
Comment (javanna) [inline x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldType.java:102]: I was wondering, would it make sense to add tests for the unsupported queries, to verify that they throw error?

File: x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/30_double.yml [hunk 2]
```diff
@@ -0,0 +1,203 @@
+---
+setup:
```
Comment (javanna) [inline x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/30_double.yml:2]: do we have tests somewhere to verify that the scripted fields work fine when used for sorting? Maybe we should also have tests for when they get referred to as part of scripts? I would not cover these two scenarios in a yaml test though.

File: x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldTypeTests.java [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+                assertThat(searcher.count(build("value(source.foo)").termsQuery(List.of(2.1, 1), mockContext())), equalTo(2));
+            }
+        }
+    }
+
+    public void testTermsQueryIsExpensive() throws IOException {
+        checkExpensiveQuery((ft, ctx) -> ft.termsQuery(List.of(randomLong()), ctx));
+    }
+
+    private ScriptDoubleMappedFieldType build(String code) throws IOException {
```
Comment (javanna) [inline x-pack/plugin/runtime-fields/src/test/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldTypeTests.java:193]: I think that most of these methods could be static, not that it matters a lot though :)

**Review by javanna — APPROVED** [review]
left a couple of suggestions for possible later improvements, LGTM though

File: x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldType.java (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/ScriptDoubleMappedFieldType.java:102]: I can do that!

File: x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/30_double.yml (hunk 2 above)
Comment (nik9000) [reply x-pack/plugin/src/test/resources/rest-api-spec/test/runtime_fields/30_double.yml:2]: We do have a scenario where a scripted field refers to a another one, but not where a script query refers to a scripted field. I'll add that.

**Comment by nik9000** [comment]
Thanks @javanna !

**Pull request merged**